### PR TITLE
homepage and site pages

### LINKED ISSUE #754: new homepage

### PR DESCRIPTION
**Pull request by gmarinov**
resolve #754 #736 #737 #702 #693 #689 #760 #756 #717 #704

 - new homepage
 - basic site template
 - asset cards
 - generic ‘scrollto’ links
 - generic svg buttons
 - other fixes and refinements

### CODE REVIEW

File: browser/scripts/ui-site.js [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+
+}
+
+jQuery('document').ready(siteUI.init);
+
+
+if (typeof VizorUI === 'undefined') var VizorUI = {};
+
+VizorUI.makeVRCanvasResizeHandler = function($playerCanvas, $containerRef) {
+	if (!($playerCanvas instanceof jQuery)) {
```
Comment (kschzt) [inline browser/scripts/ui-site.js]: when would this ever happen?

File: browser/scripts/ui-site.js [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+			OK : {
+				label: okLabel || "OK",
+				callback: function(){}
+			}
+		}
+	}
+	return VizorUI.modalOpen('<p>'+message+'</p>', heading, className, true, opts);
+}
+
+// fallback function in case E2.util is unavailable
```
Comment (kschzt) [inline browser/scripts/ui-site.js]: when is E2.util not available? don't repeat the same code

File: browser/scripts/ui-site.js [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+}
+
+/**
+ * adds handler to check whether desired username is available.
+ * this is used in the signup form, and may be used in the account details form
+ * @param $input jQuery
+ * @param currentUsername string
+ */
+VizorUI._setupAccountUsernameField = function($input, currentUsername) {
+	if (!($input instanceof jQuery)) {
```
Comment (kschzt) [inline browser/scripts/ui-site.js]: when does this ever happen?

File: browser/scripts/ui-site.js [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+		msg("ERROR: unrecognised $input");
+		return false;
+	}
+	var _t = null;
+	var lastValue=false;
+	$input.on('keyup', function(e){
+		var value = $input.val().trim();
+		var checkUsername = currentUsername || E2.models.user.username;	// last resort
```
Comment (kschzt) [inline browser/scripts/ui-site.js]: `E2.models.user.get('username')`

File: browser/scripts/ui-site.js (hunk 1 above)
Comment (gmarinov) [inline browser/scripts/ui-site.js]: usually past midnight, when one is bleary-eyed, i expect :)
also in all other times one accidentally passes just a documentElement (a `this` or `e.target` being the usual culprits)

File: controllers/graphController.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
+var _ = require('lodash')
 var Graph = require('../models/graph')
 var AssetController = require('./assetController')
 var fsPath = require('path')
 var assetHelper = require('../models/asset-helper')
 var templateCache = new(require('../lib/templateCache'))
 
+// want middleware
```
Comment (kschzt) [inline controllers/graphController.js]: what does this mean?

**Comment by kschzt** [comment]
lgtm otherwise, I'd remove the redundant checks and clarify the comments :)

File: controllers/graphController.js (hunk 5 above)
Comment (gmarinov) [inline controllers/graphController.js]: i'll waffle this ;)

**Comment by kschzt** [comment]
other than the todo comment, lgtm

**Pull request merged**